src/cluster/registration.go

```diff
@@ -51,22 +53,52 @@ func (c DeviceCluster) HandleDiscoveredDevice(newDevice model.NewDevice) error {
 		return err
 	}
 
-	if len(results) > 0 {
-		return nil
-	}
-
-	completeDevice, err := c.DeviceRepository.Create(newDevice)
-	if err != nil {
-		return err
+	var discoveredDevice *model.Device
+	switch len(results) {
+	case 0:
+		discoveredDevice, err = c.DeviceRepository.Create(newDevice)
+		if err != nil {
+			return err
+		}
+	case 1:
+		discoveredDevice = updateDeviceWithDiscoveredData(results[0], newDevice)
+		discoveredDevice.Active = true
+		err := c.DeviceRepository.Update(updateDeviceFromDevice(discoveredDevice))
+		if err != nil {
+			return err
+		}
+	default:
+		return errors.New("multiple results returned for 1 mac address")
```

Comment:
The db should block this ever happening but I suppose it doesn't hurt to have to protection

Reply:
> Tests? :)

They're coming lol

Reply:
Yea, I felt similarly but it also felt wrong to ignore the case. It could help with issues in future tests at least